Breeze.Registration/RegistrationManager.cs

```diff
@@ -142,18 +142,18 @@ public void ProcessBlock(int height, Block block)
                     }
 
                     //Check if the registration transaction has enough confirmations.
-                    if (!record.RegistrationMature && record.BlockReceived + REGISTRATION_MATURITY_BLOCK_COUNT <= height)
+                    if (!record.RegistrationMature && record.BlockReceived + WINDOW_PERIOD_BLOCK_COUNT + REGISTRATION_MATURITY_BLOCK_COUNT < height)
                     {
                         record.RegistrationMature = true;
                         this.registrationStore.AddWithReplace(record);
-                        this.logger.LogInformation($"Sufficient number of confirmations have been received for registration {record.Record.ServerId}.");
+                        this.logger.LogInformation($"Sufficient number of confirmations and collateral have been received for registration {record.Record.ServerId}.");
                     }
-                    else if (record.RegistrationMature && record.BlockReceived + REGISTRATION_MATURITY_BLOCK_COUNT > height)
+                    else if (record.RegistrationMature && record.BlockReceived + WINDOW_PERIOD_BLOCK_COUNT + REGISTRATION_MATURITY_BLOCK_COUNT >= height)
```

Comment:
I'm not sure I follow how record.RegistrationMature would be set to true to make this branch get hit

Reply:
This should work but let double check the logic
 - WINDOW_PERIOD_BLOCK_COUNT = 30
 - REGISTRATION_MATURITY_BLOCK_COUNT  = 10

if (!record.RegistrationMature && record.BlockReceived + WINDOW_PERIOD_BLOCK_COUNT + REGISTRATION_MATURITY_BLOCK_COUNT < height)
         record.RegistrationMature = true;
else
         record.RegistrationMature = false; 

1) Starting Masternode will create registration transaction (height = 10; BlockReceived = 10; RegistrationMature = false)
2) After 30 blocks we have (height = 40; BlockReceived = 10; RegistrationMature = false)
3) After one more block we have (height = 41; BlockReceived = 10; RegistrationMature = false) and if the collateral has not been yet received the registration is removed with a statement before the one we are considering now
4) After another 10 blocks and only if the collateral has been received (height = 51; BlockReceived = 10; RegistrationMature = true)